navigationrail/navigationrail/src/main/java/com/microsoft/device/dualscreen/navigationrail/NavigationRailView.kt

```diff
@@ -0,0 +1,522 @@
+        }
+
+    /**
+     * Use an animation to move the buttons when the menu gravity is changed  or [arrangeButtons] is called.
+     * By default the [AccelerateDecelerateInterpolator] is used.
+     */
+    var useAnimation: Boolean = true
+
+    /**
+     * Set the interpolator for the animation when menu gravity is changed or [arrangeButtons] is called.
+     * By default the [AccelerateDecelerateInterpolator] is used.
+     */
+    var animationInterpolator: BaseInterpolator = AccelerateDecelerateInterpolator()
+
+    /**
+     * Allows the buttons to be moved to [DisplayPosition.START] or [DisplayPosition.END] with a swipe gesture.
+     */
+    var allowFlingGesture: Boolean = true
+
+    private fun extractAttributes(context: Context, attrs: AttributeSet?) {
+        val styledAttributes =
+            context.theme.obtainStyledAttributes(
+                attrs,
+                R.styleable.ScreenManagerAttrs,
+                0,
+                0
+            )
+        try {
+            screenMode = ScreenMode.fromId(
+                styledAttributes.getResourceId(
+                    R.styleable.ScreenManagerAttrs_tools_application_mode,
+                    ScreenMode.DUAL_SCREEN.ordinal
+                )
+            )
+            useAnimation =
+                styledAttributes.getBoolean(
+                    R.styleable.ScreenManagerAttrs_useAnimation,
+                    useAnimation
+                )
+            allowFlingGesture =
+                styledAttributes.getBoolean(
+                    R.styleable.ScreenManagerAttrs_allowFlingGesture,
+                    allowFlingGesture
+                )
+        } finally {
+            styledAttributes.recycle()
+        }
+    }
+
+    private var topScreenHeight = -1
+    private var bottomScreenHeight = -1
+    private var appWindowFrameHeight = -1
+    private var hingeHeight = -1
+    private var statusBarHeight = -1
+
+    private var screenSize = Point()
+    private var appWindowPosition = Rect()
+    private var hingePosition = Rect()
+    private var shouldRedrawMenu = true
+
+    private fun setScreenParameters() {
+        if (!isIntersectingHorizontalHinge()) {
+            return
+        }
+        context.getWindowVisibleDisplayFrame().let { windowRect ->
+            appWindowPosition = windowRect
+            appWindowFrameHeight = windowRect.height()
+
+            normalizeFoldingFeatureRectForView().let { hingeRect ->
+                hingePosition = hingeRect
+                (context as Activity).windowManager.defaultDisplay.getSize(screenSize)
+
+                statusBarHeight = screenSize.y - appWindowFrameHeight
+
+                hingeHeight = hingeRect.height()
+                topScreenHeight = hingeRect.top - windowRect.top
+                bottomScreenHeight = windowRect.bottom - hingeRect.bottom
+                shouldRedrawMenu = true
+            }
+        }
+    }
+
+    private fun isIntersectingHorizontalHinge(): Boolean {
```

Comment:
This tells us if the instance of the NavigationRailView is intersecting the folding feature. 
isSeparating is used to see if the entire window should is split into multiple physical areas.
We can have the case when the isSeparating is true, but the NavigationRailView does not intersect the folding feature and thus should not split it's content.

That being said, I added the isSeparating condition to avoid the split when the folding feature is not interfering with the view.